### PR TITLE
CSS-4433: Add auth config

### PR DESCRIPTION
**Pull request by kelkawi-a**
- This PR adds auth config for the Temporal Web UI. It has been tested with Google IAM as a provider. Auth requires the ingress relation to work as it requires a hostname.
- Updated nginx ingress library to include backend protocol.

Screenshot from 2023-06-06 12-28-25

Screenshot from 2023-06-06 12-29-02

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -159,6 +162,18 @@ def _validate(self):
         if not self._state.server_status == "ready":
             raise ValueError("ui:temporal relation: server is not ready")
 
+        if not int(self.config["port"]):
```
Comment (kian99) [inline src/charm.py:165]: If we're going to do this check here you should probably make a wrapper `get_port` and use that when you initialise the nginx relation above too.

File: src/charm.py [hunk 2]
```diff
@@ -159,6 +162,18 @@ def _validate(self):
...
+            raise ValueError("Invalid config: port number missing")
+
+        if self.config["auth-enabled"]:
+            for param in REQUIRED_AUTH_PARAMETERS:
+                if self.config[param].strip() == "":
+                    raise ValueError(f"Invalid config: {param} value missing")
+
+            ingress_relation = self.model.relations["nginx-route"]
```
Comment (kian99) [inline src/charm.py:173]:
```suggestion
            ingress_relation = self.model.relations.get("nginx-route")
```

**Review by kian99 — APPROVED** [review]
lgtm with some suggestions

File: src/charm.py [hunk 3]
```diff
@@ -159,6 +162,15 @@ def _validate(self):
         if not self._state.server_status == "ready":
             raise ValueError("ui:temporal relation: server is not ready")
 
+        if self.config["auth-enabled"]:
+            for param in REQUIRED_AUTH_PARAMETERS:
+                if self.config[param].strip() == "":
+                    raise ValueError(f"Invalid config: {param} value missing")
+
+            ingress_relation = self.model.relations.get("nginx-route")
+            if not ingress_relation:
```
Comment (TeodorPt) [inline src/charm.py:171]: directly `if not self.model.relations.get("nginx-route")`?

**Review by TeodorPt — APPROVED** [review]

**Pull request merged**